src/main/java/org/opensearch/security/dlic/rest/api/AuthFailureListenersApiAction.java

```diff
@@ -0,0 +1,265 @@
+
+                        return allowedKeys.put(TYPE_JSON_PROPERTY, DataType.STRING)
+                            .put(IGNORE_HOSTS_JSON_PROPERTY, DataType.ARRAY)
+                            .put(AUTHENTICATION_BACKEND_JSON_PROPERTY, DataType.STRING)
+                            .put(ALLOWED_TRIES_JSON_PROPERTY, DataType.INTEGER)
+                            .put(TIME_WINDOW_SECONDS_JSON_PROPERTY, DataType.INTEGER)
+                            .put(BLOCK_EXPIRY_JSON_PROPERTY, DataType.INTEGER)
+                            .put(MAX_BLOCKED_CLIENTS_JSON_PROPERTY, DataType.INTEGER)
+                            .put(MAX_TRACKED_CLIENTS_JSON_PROPERTY, DataType.INTEGER)
+                            .build();
+                    }
+                });
+            }
+        };
+    }
+
+    private ToXContent authFailureContent(final ConfigV7 config) {
+        return (builder, params) -> {
+            builder.startObject();
+            for (String name : config.dynamic.auth_failure_listeners.getListeners().keySet()) {
+                ConfigV7.AuthFailureListener listener = config.dynamic.auth_failure_listeners.getListeners().get(name);
+                builder.startObject(name);
+                builder.field(NAME_JSON_PROPERTY, name)
+                    .field(TYPE_JSON_PROPERTY, listener.type)
+                    .field(IGNORE_HOSTS_JSON_PROPERTY, listener.ignore_hosts)
+                    .field(AUTHENTICATION_BACKEND_JSON_PROPERTY, listener.authentication_backend)
+                    .field(ALLOWED_TRIES_JSON_PROPERTY, listener.allowed_tries)
+                    .field(TIME_WINDOW_SECONDS_JSON_PROPERTY, listener.time_window_seconds)
+                    .field(BLOCK_EXPIRY_JSON_PROPERTY, listener.block_expiry_seconds)
+                    .field(MAX_BLOCKED_CLIENTS_JSON_PROPERTY, listener.max_blocked_clients)
+                    .field(MAX_TRACKED_CLIENTS_JSON_PROPERTY, listener.max_tracked_clients);
+                builder.endObject();
+            }
+            builder.endObject();
+            return builder;
+        };
+    }
+
+    private void authFailureConfigApiRequestHandlers(RequestHandler.RequestHandlersBuilder requestHandlersBuilder) {
+
+        requestHandlersBuilder.override(
+            GET,
+            (channel, request, client) -> loadConfiguration(getConfigType(), false, false).valid(configuration -> {
+                final var config = (ConfigV7) configuration.getCEntry(CType.CONFIG.toLCString());
+                ok(channel, authFailureContent(config));
+            }).error((status, toXContent) -> response(channel, status, toXContent))
+        ).override(DELETE, (channel, request, client) -> loadConfiguration(getConfigType(), false, false).valid(configuration -> {
+            ConfigV7 config = (ConfigV7) configuration.getCEntry(CType.CONFIG.toLCString());
+
+            String listenerName = request.param("name");
```

Comment:
nit: Replace this with the constant, similar to below

Reply:
Done in: [2a8c237](https://github.com/opensearch-project/security/pull/4641/commits/2a8c237f25239167cf854bc50143f9f7f85a4af9)